streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -50,11 +51,12 @@
     static final String LOCK_FILE_NAME = ".lock";
     private static final Logger log = LoggerFactory.getLogger(StateDirectory.class);
 
+    private final Time time;
+    private final String appId;
     private final File stateDir;
     private final boolean createStateDirectory;
```

Comment:
What do you think about renaming this, eg  something like `isStateful, `hasStatefulTopology`, etc?

It's really confusing to reason about in its usage, for example why `createStateDirectory => #lock returns true`.  You have to backtrack to where `createStateDirectory` is set to understand this. However, I think it's easy to reason about why `isStateful => should create state directory`.
 
Alternatively, what if we have a stateless version of the `StateDirectory` class that just stubs things where appropriate. Then we could get rid of this altogether. WDYT?

Reply:
That makes sense, I will do the renaming.